src/index.js

```diff
@@ -1,108 +1,120 @@
-type StyledElementPropsType = {
-  classes: Object,
-  children: ?any,
-  className: ?string,
-}
 
-const createStyled = (jss?: Function = jssDefault) => (baseStyles: Object = {}) => {
-  let sheet
-  let dynamicSheet
+const createStyled = (
+  jss?: Function = jssDefault
+) => (
+  baseStyles: BaseStylesType = {}
+): StyledType => {
+  const sheets = {}
   let counter = 0
 
+  const getScopedTagName = (tagName: string) => `${tagName}-${++counter}`
+
+  const mountSheets = () => {
+    if (!sheets.staticSheet) {
+      sheets.staticSheet = jss.createStyleSheet(baseStyles, {
+        meta: 'StaticBaseSheet',
+      }).attach()
+
+      sheets.dynamicSheet = jss.createStyleSheet({}, {
+        link: true,
+        meta: 'DynamicComponentSheet',
+      }).attach()
+    }
+  }
+
   const styled = (
-    tagOrStyledElement: tagOrStyledElementTypeype,
-    ownStyles: Object
+    tagNameOrStyledElement: TagNameOrStyledElementType,
+    ownStyle: ComponentStyleType
   ): StyledElementType => {
-    const {tag, styles}: StyledElementAttrsType = typeof tagOrStyledElement === 'string'
-      ? {tag: tagOrStyledElement, styles: {}}
-      : tagOrStyledElement
+    const {tagName, style}: StyledElementAttrsType = typeof tagNameOrStyledElement === 'string'
+      ? {tagName: tagNameOrStyledElement, style: {}}
+      : tagNameOrStyledElement
 
-    const elementStyles = {...styles, ...ownStyles}
-    const dynamicStyles = getDynamicStyles(elementStyles)
-    const staticTag = `${tag}-${++counter}`
+    const elementStyle = {...style, ...ownStyle}
+    const dynamicStyle = getDynamicStyles(elementStyle)
+    const staticTagName = getScopedTagName(tagName)
 
     return class StyledElement extends PureComponent {
-      static tag = tag
-
-      static styles = elementStyles
+      static tagName: string = tagName
+      static style: ComponentStyleType = elementStyle
 
       props: StyledElementPropsType
 
-      tagScoped = ''
+      dynamicTagName = ''
 
       constructor(props) {
         super(props)
-        this.tagScoped = `${tag}-${++counter}`
+        this.dynamicTagName = getScopedTagName(tagName)
       }
 
       componentWillMount() {
-        if (!sheet) {
-          sheet = jss.createStyleSheet(baseStyles, {
-            link: true,
-            meta: 'StaticBaseSheet',
-          }).attach()
-
-          dynamicSheet = jss.createStyleSheet({}, {
-            link: true,
-            meta: 'DynamicComponentSheet',
-          }).attach()
-        }
+        mountSheets()
 
-        if (!sheet.getRule(staticTag)) {
-          sheet.addRule(staticTag, elementStyles)
+        if (!sheets.staticSheet.getRule(staticTagName)) {
+          sheets.staticSheet.addRule(staticTagName, elementStyle)
         }
 
-        if (dynamicStyles && !dynamicSheet.getRule(this.tagScoped)) {
-          dynamicSheet
+        if (dynamicStyle && !sheets.dynamicSheet.getRule(this.dynamicTagName)) {
+          sheets.dynamicSheet
```

Comment:
do we reassign them somewhere else than in `mountSheets?`? Should evtl mountSheets return them?